### ISSUE TITLE
Make the spine demos more portable and reusable

### ISSUE THREAD

**Issue by jake** [EsotericSoftware/spine-runtimes#1011]
The [Spine Runtime Demos](http://esotericsoftware.com/spine-demos) page has a lot of potentially useful code, but it's very hard to discover or reuse.

You [asked me](https://twitter.com/EsotericSoft/status/916307284459950081) on Twitter to define what I meant by "modular code". I just mean more self contained — something you can copy and paste out of the demo page and have work.

Currently, the demo page has a whole "demo framework", which manages all the `<canvas>`'s / etc, that is very specific to your spine-demos page. It makes the code very interdependent. It took me a while to rip out all the unrelated code, and I had to jump over to the repo to even find it.

Simply showing some annotated code would be a start:
https://framer.com/docs/
https://developers.google.com/maps/

Even nicer would be to build something [Codepen](https://codepen.io/) inspired. On each demo, add a "Source" tab that you can switch to and get a chunk of HTML and JS to recreate that demo.

**Comment by badlogic** [comment]
Thanks for the follow up. I guess the biggest problem is, that the spine-ts WebGL runtime isn't an engine per se, but instead a framework with a few utility classes for doing common things such as reacting to mouse/touch/keyboard etc. to be re-used by engine creators like pixi.js.

The most bare-bones example is this https://github.com/EsotericSoftware/spine-runtimes/blob/3.6/spine-ts/webgl/example/test.html It renderers the same skeleton a bunch of times at random locations. 

As you can see, there's still a lot of management code (asset loading, shaders, etc.) due to the fact that we do not aim to provide a full engine, but rather building blocks to be integrated in engines.

Usually users resort to either our spine-ts Widget for displaying a single skeleton in a single canvas (mostly artists wanting to demo their art on their website), or they use an engine like pixi.js which integrates the Spine runtime.

The Spine runtime API is the same across the different language implementations. Our runtime guide describes the API and its concepts similar to the two doc sites you linked to, including code samples: http://esotericsoftware.com/spine-runtimes-guide. 

What's currently lacking is the documentation of the WebGL specific APIs such as AssetManager, Shader, etc. Since the large majority of users consumes the spine-ts runtime via an engine like pixi.js, we haven't put many resources into documenting that part of the API yet. I'm happy to do so, if there is a need though!

As for untangling the demos and make them similar to Codepen, we have considered this. However, given the number of spine-ts users, we have not committed to doing this yet, as the resources needed to implement it.

To summarize, I'll look into extending the documentation to cover the WebGL specific API, I'll add more "bare bone" examples like the one linked above. We currently can not commit resources to implementing something like Codepen for the Spine demos.
